api/src/controllers/user.controller.js

```diff
@@ -1,51 +1,81 @@
-import jwt from 'jsonwebtoken';
 import bcrypt from 'bcryptjs';
 import UserService from '../services/user.service';
 
-const secret = process.env.SECRET || 'supersecret';
 
 const UserController = {
-  fetchAllUsers(req, res) {
-    const allUsers = UserService.fetchAllUsers();
-    // eslint-disable-next-line array-callback-return
-    allUsers.map((userObj) => {
-      Object.defineProperty(userObj, 'password', {
-        enumerable: false,
-        writable: true,
-      });
-    });
-    return res.json({
-      status: 200,
-      data: allUsers,
-    }).status(200);
-  },
+  // fetchAllUsers(req, res) {
+  //   const allUsers = UserService.fetchAllUsers();
+  //   // eslint-disable-next-line array-callback-return
+  //   allUsers.map((userObj) => {
+  //     Object.defineProperty(userObj, 'password', {
+  //       enumerable: false,
+  //       writable: true,
+  //     });
+  //   });
+  //   return res.json({
+  //     status: 200,
+  //     data: allUsers,
+  //   }).status(200);
+  // },
 
-  addAUser(req, res) {
+  async addAUser(req, res) {
     const newUser = req.body;
     if (!newUser.type && !newUser.isAdmin) {
       newUser.type = 'client';
       newUser.isAdmin = false;
     }
+    if (!newUser.firstName) {
```

Comment:
can this be done with express validator or joi and it should be on the route. Write a validation middleware and call it on the route and not on the controller.

Reply:
Done. I used joi